examples/blend/main.go

```diff
@@ -0,0 +1,218 @@
+		{blend: ebiten.BlendDestinationOut, name: "BlendDestinationOut"},
+		{blend: ebiten.BlendClear, name: "BlendClear"},
+		{blend: ebiten.BlendXor, name: "BlendXor"},
+		{blend: ebiten.BlendLighter, name: "BlendLighter"},
+	}
+
+	// Adjusting the tile size for the available images.
+	g.tileSize = maxSide(g.source, g.dest)
+	g.offscreen = ebiten.NewImage(int(g.tileSize), int(g.tileSize))
+
+	return g, nil
+}
+
+func (g *Game) Update() error {
+	return nil
+}
+
+func (g *Game) Draw(screen *ebiten.Image) {
+	screen.Fill(color.White)
+
+	// Get an offset for center alignment.
+	// Where sw, sh is the screen size in pixels.
+	// And gw, gh is the grid size in pixels.
+	sw, sh := float64(screen.Bounds().Dx()), float64(screen.Bounds().Dy())
+	gw, gh := g.gridW*(g.tileSize+g.tileSpacing)-g.tileSpacing, g.gridH*(g.tileSize+g.tileSpacing)
+	cx, cy := (sw-gw)/2, (sh-gh)/2
+
+	// Drawing a tilemap
+	for y := 0.; y < g.gridH; y++ {
+		for x := 0.; x < g.gridW; x++ {
+			px, py := x*(g.tileSize+g.tileSpacing), y*(g.tileSize+g.tileSpacing)
+			if y > 0 {
+				// Making a place for the text.
+				py += g.textSpacing * y
+			}
+
+			// Getting the right mode for the coords or skipping the rendering.
+			m, ok := getMode(g.modes, int(x), int(y), int(g.gridW), int(g.gridH))
+			if !ok {
+				continue
+			}
+
+			// Drawing the blend mode and it's name.
+			g.drawBlendMode(screen, px+cx, py+cy, m.blend)
+			drawCenteredText(screen, px+cx+g.tileSize/2, py+cy+g.tileSize+g.textSpacing, m.name)
+		}
+	}
+}
+
+func (g *Game) Layout(w, h int) (int, int) {
+	return w, h
+}
+
+func (g *Game) drawBlendMode(screen *ebiten.Image, x, y float64, mode ebiten.Blend) {
+	// Copy the destination image to offscreen so as not to modify it.
+	g.offscreen.Clear()
+	g.offscreen.DrawImage(g.dest, nil)
+
+	// Select and apply blending mode.
+	op := &ebiten.DrawImageOptions{}
+	op.Blend = mode
+	g.offscreen.DrawImage(g.source, op)
+
+	// Draw the result on the passed coordinates.
+	op = &ebiten.DrawImageOptions{}
+	op.GeoM.Translate(x, y)
+	screen.DrawImage(g.offscreen, op)
+}
+
+// loadImage is a util function for loading embedded images.
+func loadImage(data []byte) (*ebiten.Image, error) {
+	m, _, err := image.Decode(bytes.NewReader(data))
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode image: %w", err)
+	}
+	return ebiten.NewImageFromImage(m), nil
+}
+
+// maxSide is a util function for adjusting the tile size for the available images.
+func maxSide(a, b *ebiten.Image) float64 {
+	xMax := math.Max(float64(a.Bounds().Dx()), float64(a.Bounds().Dx()))
+	yMax := math.Max(float64(b.Bounds().Dx()), float64(b.Bounds().Dx()))
+	return math.Max(xMax, yMax)
+}
+
+// index is a util function that returns 1D index from 2D.
+func index(x, y, w int) int {
+	return y*w + x
+}
+
+// inRange is a util function to check if the index is in range.
+func inRange(x, y, w, h int) bool {
+	xOut := x < 0 || x >= w
+	yOut := y < 0 || y >= h
+	if xOut || yOut {
+		return false
+	}
+	return true
```

Comment:
`return 0 <= x && x < w && 0 <= y && y < h`

As this function is called only once, I think this doesn't have to be a separate function.

Reply:
Only to improve readability with early return

Reply:
I mean, that I will need additional "if" for this in getMode func.

Reply:
> I mean, that I will need additional "if" for this in getMode func that I previously had in separate function.

I'm afraid I still don't understand what you said, but if we remove this function, getMode will be like this:

```go
func getMode(modes []mode, x, y, w, h int) (m mode, ok bool) {
	if x < 0 || x >= w || y < 0 || y >= h {
		return mode{}, false
	}
	if index(x, y, w) > len(modes)-1 {
		return mode{}, false
	}
	return modes[idx], true
}
```

> I'm talking about the first return in this func.

I prefer one-line implementation as I said, if we keep this function

```
return 0 <= x && x < w && 0 <= y && y < h
```

Reply:
So I'm fine with an additional `if`.

Reply:
Ok, I'll inline InRange.

Reply:
Fixed.